cmd/queue/main.go

```diff
@@ -192,6 +192,11 @@ func main() {
 		flush(logger)
 		os.Exit(1)
 	case <-ctx.Done():
+		// If container-freezer enabled, resume container so that normal shutdown can occur
+		if env.ConcurrencyStateEndpoint != "" {
+			ce := queue.NewConcurrencyEndpoint(env.ConcurrencyStateEndpoint, env.ConcurrencyStateTokenPath)
```

Comment:
presumably this will end up independently reloading the token which seems bad, can we arrange for the same concurrency endpoint as elsewhere to be used here?

Reply:
It's just reading the existing token from the mounted volume, so I _think_ it's ok to read it again here (i.e. it shouldn't change the actual token value). That said, if we go with the shutdown hook idea from the previous comment rather than calling the freezer here, that probably renders this moot anyway.

Reply:
I mean it'll work, but we'll have two loops and two reads of the file which seems at least unfortunate/confusing :)

Reply:
Fair point. I was aiming to avoid having to pass the concurrency endpoint around, but maybe that's the better way to go about it (similar to what's done with the `drainer`), though then we're also starting to leak... let me see if we can avoid all of this with a shutdown hook.

Reply:
to be clear when I said "shutdown hook" I just mean adding ce.Shutdown alongside ce.Pause/ce.Resume